chore: deprecate `tensorflow` maintenance support (#7989)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Deprecations 👋</h3>

* Maintenance support for the ``tensorflow`` interface has been deprecated and will be dropped in Pennylane v0.44.
  Future versions of PennyLane are not guaranteed to work with TensorFlow.
  Instead, we recommend using the :doc:`jax </introduction/interfaces/jax>` or :doc:`torch </introduction/interfaces/torch>` interface for

>   machine learning applications to benefit from enhanced support and features.

Reviewer: Same comment as above re: demos instead of this doc page

Author: Done in https://github.com/PennyLaneAI/pennylane/pull/7989/commits/fdd83aca08d48d2a533729de2ef355cd57448282. Waiting for render!

Author: Pushed a fix so it says JAX and PyTorch for the first set of hyperlinks.

<h3>Deprecations 👋</h3>

* Support for using TensorFlow with PennyLane has been deprecated and will be dropped in Pennylane v0.44.
  Future versions of PennyLane are not guaranteed to work with TensorFlow.
  Instead, we recommend using the :doc:`JAX </introduction/interfaces/jax>` or :doc:`PyTorch </introduction/interfaces/torch>` interface for
  machine learning applications to benefit from enhanced support and features. Please consult the following demos for 
  a comprehensive guide on JAX and PyTorch: 
  [Turning quantum nodes into Torch Layers](https://pennylane.ai/qml/demos/tutorial_qnn_module_torch) and 
  [How to optimize a QML model using JAX and Optax](https://pennylane.ai/qml/demos/tutorial_How_to_optimize_QML_model_using_JAX_and_Optax).

>   [(#7989)](https://github.com/PennyLaneAI/pennylane/pull/7989)

Author: https://github.com/PennyLaneAI/pennylane/pull/7989/commits/1cbff2e2a5d8bfa0596bd784b4e273400a6548fe 🤞🏼

>   [(#7989)](https://github.com/PennyLaneAI/pennylane/pull/7989)

Author: Hey so, I don't think these sphinx directives are working. I'm looking at old change logs and it looks like we just hard linked the demos (  [KAK decomposition](https://pennylane.ai/qml/demos/tutorial_kak_decomposition) ) for example. Can we continue with this?

<h3>Deprecations 👋</h3>

* Support for using TensorFlow with PennyLane has been deprecated and will be dropped in Pennylane v0.44.
  Future versions of PennyLane are not guaranteed to work with TensorFlow.
  Instead, we recommend using the :doc:`JAX </introduction/interfaces/jax>` or :doc:`PyTorch </introduction/interfaces/torch>` interface for

>   machine learning applications to benefit from enhanced support and features. Please consult the following demos for 
>   a comprehensive guide on
>   :doc:`PyTorch </qml/demos/tutorial_qnn_module_torch>` or
>   :doc:`JAX </qml/demos/tutorial_How_to_optimize_QML_model_using_JAX_and_Optax>`.

Reviewer: <img width="872" height="157" alt="Screenshot 2025-08-12 at 8 56 29 AM" src="https://github.com/user-attachments/assets/6aae00e9-2a14-45a5-b661-14765393bb29" />

I'm so confused—I thought this worked ❓ 

Anyway, hard-coded links it is. 

```suggestion
  machine learning applications to benefit from enhanced support and features. Please consult the following demos for
  more usage information: 
  `Turning quantum nodes into Torch Layers <https://pennylane.ai/qml/demos/tutorial_qnn_module_torch>` and
  `How to optimize a QML model using JAX and Optax <https://pennylane.ai/qml/demos/tutorial_How_to_optimize_QML_model_using_JAX_and_Optax>`.
  [(#7989)](https://github.com/PennyLaneAI/pennylane/pull/7989)
```

<h3>Deprecations 👋</h3>

* Support for using TensorFlow with PennyLane has been deprecated and will be dropped in Pennylane v0.44.
  Future versions of PennyLane are not guaranteed to work with TensorFlow.
  Instead, we recommend using the :doc:`JAX </introduction/interfaces/jax>` or :doc:`PyTorch </introduction/interfaces/torch>` interface for
  machine learning applications to benefit from enhanced support and features. Please consult the following demos for
  more usage information: 

>   `Turning quantum nodes into Torch Layers <https://pennylane.ai/qml/demos/tutorial_qnn_module_torch>` and
>   `How to optimize a QML model using JAX and Optax <https://pennylane.ai/qml/demos/tutorial_How_to_optimize_QML_model_using_JAX_and_Optax>`.

Reviewer: ```suggestion
  [Turning quantum nodes into Torch Layers](https://pennylane.ai/qml/demos/tutorial_qnn_module_torch) and
  [How to optimize a QML model using JAX and Optax](https://pennylane.ai/qml/demos/tutorial_How_to_optimize_QML_model_using_JAX_and_Optax).
```

  [(#7989)](https://github.com/PennyLaneAI/pennylane/pull/7989)

>   [(#7989)](https://github.com/PennyLaneAI/pennylane/pull/7989)

Author: ```suggestion
```
